license/src/com/untangle/app/license/LicenseManagerImpl.java

```diff
@@ -1074,21 +1091,30 @@ private String _getLicenseUrl()
     }
 
     /**
-     * syncs the license server state with local state
+     * Syncs the license server state with local state
      */
     private void _syncLicensesWithServer()
     {
         logger.info("Reloading licenses..." );
 
         synchronized (LicenseManagerImpl.this) {
+            boolean downloadChanged = false;
+            boolean revocationsChanged = false;
             _readLicenses();
 
             if ((! UvmContextFactory.context().isDevel()) || (devLicenseTest)) {
-                _downloadLicenses();
-                _checkRevocations();
+                downloadChanged = _downloadLicenses();
```

Comment:
since we return settings changed here, does it make sense to call saveSettings in a single call after this?

Reply:
ie: where we have if(downloadChanged ||revocationsChanged) _setSettings once, instead of the potential two times above?